src/ray/core_worker/task_execution.cc

```diff
@@ -1,7 +1,81 @@
 #include "task_execution.h"
+#include "context.h"
+#include "core_worker.h"
 
 namespace ray {
 
-void CoreWorkerTaskExecutionInterface::Start(const TaskExecutor &executor) {}
+Status CoreWorkerTaskExecutionInterface::Run(const TaskExecutor &executor) {
+  RAY_CHECK(core_worker_.is_initialized_);
+
+  while (true) {
+    std::unique_ptr<raylet::TaskSpecification> task_spec;
+    // RAY_CHECK_OK(core_worker_.raylet_client_->GetTask(&task_spec));
+    auto status = core_worker_.raylet_client_->GetTask(&task_spec);
+    if (!status.ok()) {
+      RAY_LOG(ERROR) << "Get task failed with error: "
+                     << ray::Status::IOError(status.message());
+      return status;
+    }
+
+    auto &spec = *task_spec;
```

Comment:
As discussed with @raulchen in the past, I think it would add clarity to make as many things `const` as possible (so that readers know when things can't change). See https://github.com/ray-project/ray/pull/4246#discussion_r271140269

Reply:
I see, will update. thanks.